soc/arm/Kconfig

```diff
@@ -34,7 +34,7 @@ config CPU_HAS_NRF_IDAU
 
 config CPU_HAS_FPU_DOUBLE_PRECISION
 	bool
-	depends on CPU_CORTEX_M7
+	depends on (CPU_CORTEX_M7 || CPU_CORTEX_R4 || CPU_CORTEX_R5)
```

Comment:
Now that double precision is not a single exception, but a list of exceptions, let's remove this line.

This has the added benefit of allowing out-of-tree CPUs to select it as well.